### DISCUSSION TITLE
Thank you and some Suggestions

### DISCUSSION THREAD

**Discussion by Taps7734** [porrey/Virtual-ZPL-Printer#5] [Ideas]
First I want to say thank you for this application.  I was trying to use a chrome plugin that had many issues.  Especially since Chrome isn't our corporate browser.  And trying to get it to work in Edge was a nightmare.  This standalone application is a dream. 

I do have a few very minor suggestions.

1 - Is it possible to add a way to rotate the images directly from the application?  If I open the image up and rotate it in the Windows Photos app, and then refresh the application, it show correctly.  But it could save some time by doing it directly in the app.

2 - We currently have our ERP print some labels that are actually 2 pages.  Technically it prints 2 labels.  But in this app, it only saves/shows the first page.  And then save them as zpl-label-image-1A.png and zpl-label-image-1B.png or something similar.  Had the same issue in the chrome plugin I mentioned earlier... except every once in a while it would SKIP the first page/label and just do the second.

**Comment by porrey** [comment]
I would be willing to consider both enhancements. Can you provide sample ZPL for both 1 and 2 and any details necessary to reproduce? I will need to be able run both scenarios for testing.

**Comment by porrey** [comment]
Both changes proved to be straightforward. I uploaded version 1.1.17 with these new features.

**Comment by Taps7734** [comment]
That was amazing turn around.  Thank you. You got the changes in before I was able to get you the requested info.

The rotation part is awesome.  Works like a charm.

The multiple label part has a little problem.  Don't know if it was they way you sent the info to Labelary, or if it was the way our ERP sent the data.  But when sending a 2 page label, I get the 2 labels... and then I get 5 more that say 
"LABELARY ERROR.  
ERROR: You requested the 3rd label, but the specified ZPL only generates 2 labels."

I would send you the ZPL code from my logs, but I would prefer to not post it in a public forum since it has company info in it.

Is there an email or other way that I could get them to you?

**Reply by porrey** [reply]
send to 08chases-spacey@icloud.com

**Reply by Taps7734** [reply]
Sent

**Comment by Taps7734** [comment]
I did a little googling and found that ZPL uses  "^XZ" to finish a page and "^XA" to start a page.

Do not know if that is useful for you, but I figured I offer the info just in case.

**Reply by porrey** [reply]
Thanks. 

It turns out that Labelary defaults to the first label, but a parameter can be passed to request each page. I updated the code to count the occurrences of ^XA an then call the service for each page.

**Comment by porrey** [comment]
I posted a new version that uses Labelary's HTTP header to get the label count.

**Comment by Taps7734** [comment]
Just installed 1.0.18.  Everything works well.  Thanks for your quick responses and your time.
